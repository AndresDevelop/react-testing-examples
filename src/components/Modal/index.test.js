import React from 'react';
import {
  render,
  fireEvent,
} from 'react-testing-library';
import 'jest-dom/extend-expect';
import { Modal } from './index';

describe('Modal', () => {
  it('should open and close modal', () => {
    const { container, getByText, queryByTestId, getByTestId } = render(<Modal />);

    const open = getByText('Open');
    expect(queryByTestId('modal')).not.toBeInTheDocument();

    fireEvent.click(open);

    expect(getByTestId('modal')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})
