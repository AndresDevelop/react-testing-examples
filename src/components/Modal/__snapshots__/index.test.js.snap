// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should open and close modal 1`] = `
<div>
  <button
    type="button"
  >
    Open
  </button>
  <div
    class="overlay"
  >
    <div
      class="modal"
      data-testid="modal"
    >
      <button
        type="button"
      >
        Ã—
      </button>
      <p>
        Hello Modal!
      </p>
    </div>
  </div>
</div>
`;
